lib/galaxy/model/__init__.py

```diff
@@ -2572,7 +2572,7 @@ def __create_version__(self, session):
 
         changes = {}
 
-        for attr in class_mapper(HistoryDatasetAssociation).column_attrs:
+        for attr in state.attrs:
```

Comment:
Looks like that was causing the issues, at least the integration tests passed in time.